File: website/html/java/MongoAndRAClient/src/com/cra/csfvRaRest/schemas/responses/CreatePrincipalResponse.java
/////////////////////////////////////////////////////////////////////////////// 
// Charles River Analytics, Inc., Cambridge, Massachusetts 
// Copyright (C) 2013. All Rights Reserved. 
// See http://www.cra.com or email info@cra.com for more information. 
/////////////////////////////////////////////////////////////////////////////// 
// Author: wdorin@cra.com
/////////////////////////////////////////////////////////////////////////////// 

package com.cra.csfvRaRest.schemas.responses;

import com.google.gson.annotations.Expose;

public class CreatePrincipalResponse {

  @Expose
  public Boolean success;
  @Expose
  public Boolean timeout;
  @Expose
  public String principalType;
  @Expose
  public String id;

  public CreatePrincipalResponse(Boolean success, Boolean timeout, String typeCreated, String id) {
    this.success = success;
    this.timeout = timeout;
    principalType = typeCreated;
    this.id = id;

  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("{ ");
    sb.append("\"success\" : " + success + ", ");
    sb.append("\"timeout\" : " + timeout + ", ");
    if(principalType != null)
    	sb.append("\"principalType\" : \"" + principalType.toString() + "\", ");
    sb.append("\"playerId\" : " + id + " } ");
    return sb.toString();
  }
}

File: website/html/java/MongoAndRAClient/JSON_schemas/CreatePrincipalResponse.js
/**
 * The Create Principal Response Schema
 *
 * This is the RA response to an create principal request
 *
 * @author wdorin, CRA
 * @version 1.0
 */
var SetPrincipalMetadataResponseSchema = new Schema({
    success			: { type : Boolean, required : true},
	timeout			: { type : Boolean, required : true},
	principalType	: { type : String, required : true},
    id				: { type : String, required: true }
});